vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -1668,17 +1715,51 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
                 member x.UpdateProjectCfg_Done(pHierProj, _pCfgProj, _pCfgSln, _dwAction, _fSuccess, _fCancel) =
                     UpdateConfig(pHierProj)
                     VSConstants.S_OK
+
             interface IVsUpdateSolutionEvents3 with
                 member x.OnBeforeActiveSolutionCfgChange(_oldCfg, _newCfg) =
-                    queuedWork <- Some( [] )
+                    // this will be called for each project, but wait dialogs cannot 'stack'
+                    // i.e. if a wait dialog is already open, subsequent calls to StartWaitDialog
+                    // will not override the current open dialog
+                    waitDialog <-
```

Comment:
Is there a guarantee that OnBeforeActiveSolutionCfgChange can be called multiiple times simultaneously?